dev/plugins/hu.elte.txtuml.validation/src/hu/elte/txtuml/validation/sequencediagram/SequenceDiagramCompilationParticipant.java

```diff
@@ -0,0 +1,26 @@
+package hu.elte.txtuml.validation.sequencediagram;
+
+import org.eclipse.jdt.core.dom.CompilationUnit;
+
+import hu.elte.txtuml.validation.common.AbstractJtxtUMLCompilationParticipant;
+import hu.elte.txtuml.validation.common.ProblemCollector;
+import hu.elte.txtuml.validation.sequencediagram.visitors.SequenceDiagramVisitor;
+import hu.elte.txtuml.validation.sequencediagram.visitors.Utils;
+
+public class SequenceDiagramCompilationParticipant extends AbstractJtxtUMLCompilationParticipant {
+
+	public static final String MARKER_TYPE = "hu.elte.txtuml.validation.sequencediagrammarker"; //$NON-NLS-1$
+
+	@Override
+	protected String getMarkerType() {
+		return MARKER_TYPE;
+	}
+
+	@Override
+	protected void validate(CompilationUnit unit, ProblemCollector collector) {
+		if (Utils.isSequenceDiagramDescription(unit)) {
```

Comment:
I don't think this is necessary here, as the visitor already filters sequence classes. Note that in the case of model errors, the corresponding `isModelElement` check is actually needed, because there we have to look for invalid type declarations inside model packages as well.